IdentityCore/src/pop_manager/MSIDDevicePopManager.m

```diff
@@ -0,0 +1,244 @@
+        }
+        
+        return nil;
+    }
+    
+    return keyData;
+}
+
+- (NSString *)getPublicKeyJWK
+{
+    NSString* jwk = [NSString stringWithFormat:jwkTemplate,
+                     [self getPublicKeyExp:self.publicKeyData],
+                     [self getPublicKeyMod:self.publicKeyData]];
+    
+    NSData *jwkData = [jwk dataUsingEncoding:NSUTF8StringEncoding];
+    NSData *hashedData = [jwkData msidSHA256];
+    NSString *base64EncodedJWK = [hashedData msidBase64UrlEncodedString];
+    return base64EncodedJWK;
+}
+
+- (NSString *)getRequestConfirmation:(NSError **)error
+{
+    NSString *kid = [NSString stringWithFormat:kidTemplate, [self getPublicKeyJWK]];
+    NSData *kidData = [kid dataUsingEncoding:NSUTF8StringEncoding];
+    return [kidData msidBase64UrlEncodedString];
+}
+
+- (NSString *)createSignedAccessToken:(NSString *)accessToken
+                           httpMethod:(NSString *)httpMethod
+                           requestUrl:(NSString *)requestUrl
+                                nonce:(NSString *)nonce
+                                error:(NSError **)error
+{
+    NSString *kid = [self getPublicKeyJWK];
+    
+    if (!kid)
+    {
+        if (error)
+        {
+           NSString *errorDescription = [NSString stringWithFormat:@"Error generating kid from public key bits."];
+           *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, errorDescription, nil, nil, nil, nil, nil, NO);
+        }
+        
+        return nil;
+    }
+    
+    NSURL *url = [NSURL URLWithString:requestUrl];
+    if (!url)
+    {
+        if (error)
+        {
+          NSString *errorDescription = [NSString stringWithFormat:@"Invalid request url %@", requestUrl];
+          *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, errorDescription, nil, nil, nil, nil, nil, NO);
+            
+        }
+               
+        return nil;
+    }
+    
+    NSString *host = url.host;
+    NSString *path = url.path;
+    
+    NSDictionary *header = @{
+                             @"alg" : @"RS256",
+                             @"typ" : @"JWT",
+                             @"kid" : kid
+                             };
+    
+    NSDictionary *payload = @{
+                              @"at" : accessToken,
+                              @"cnf": @{
+                                      @"jwk":@{
+                                          @"kty" : @"RSA",
+                                          @"n" : [self getPublicKeyMod:self.publicKeyData],
+                                          @"e" : [self getPublicKeyExp:self.publicKeyData]
+                                      }
+                              },
+                              @"ts" : [NSString stringWithFormat:@"%lu", (long)[[NSDate date] timeIntervalSince1970]],
+                              @"m" : httpMethod,
+                              @"u" : host,
+                              @"p" : path,
```

Comment:
this will crash if path or host is nil

Reply:
Good catch. Addressed. Updated code to fill error and return nil if host or path is nil.